kern/bpf/bpf_helpers.h

```diff
@@ -40,6 +40,9 @@
 #undef __always_inline
 #define __always_inline inline __attribute__((always_inline))
 
+#if defined(NOCORE) && defined(noinline)
```

Comment:
I don't think it's a clang version related issue.

I think, it's because `#define   noinline                      __attribute__((__noinline__))` in `include/linux/compiler_attributes.h` to cause `__noinline` expanding to `__attribute__((__attribute__((__noinline__))))`. But, clang does not support `__attribute__((__attribute__((__noinline__))))`.

Reply:
> the version greater than 15 will only have a conflict

It's OK to build with clang-18. I've tried it.